version: "3"
volumes:
  postgis-data:
  chaches-data:
services:
#  main:
#    image: main
#    networks:
#      - default
#    volumes:
#      - ./logs:/app/app-log
#      - ./config.yaml:/app/config.yaml
#    ports:
#      - "8080:8080"
#    depends_on:
#      - quests-bd
#    restart: on-failure
  banner-bd:
    image: postgres:16
    expose:
      - "5432"
    command: "postgres -c shared_preload_libraries='pg_stat_statements'"
    volumes:
      - postgis-data:/var/lib/postgresql
      - ./script:/docker-entrypoint-initdb.d
    env_file:
      - ./env/db.env
    ports:
      - "5432:5432"
    restart: on-failure
    healthcheck:
      test: "exit 0"
    deploy:
      resources:
        limits:
          cpus: '2'
          memory: 4G
  chaches:
    image: "redis:alpine"
    expose:
      - "6379"
    ports:
      - "6379:6379"
    volumes:
      - chaches-data:/data
    restart: always
networks:
  default:
    driver: bridge